src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtPostcodeController.java

```diff
@@ -0,0 +1,107 @@
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<String>> getCourtPostcodes(@PathVariable String slug) {
+        return ok(adminService.getCourtPostcodesBySlug(slug));
+    }
+
+    /**
+     * Adds new postcodes to be handled by the court for civil service type (
+     * @param slug Court slug
+     * @param postcodes a list of postcodes to be added
+     * @return A list of postcodes created if successful, else return a list of invalid postcodes and a '400' response
+     */
+    @PostMapping(path = "/{slug}/postcodes")
+    @ApiOperation("Add new court postcodes")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Postcodes created", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 400, message = "Invalid postcodes", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role(FACT_SUPER_ADMIN)
+    public ResponseEntity<List<String>> addCourtPostcodes(@PathVariable String slug, @RequestBody List<String> postcodes) {
+        final List<String> invalidPostcodes = validationService.validatePostcodes(postcodes);
+        if (CollectionUtils.isEmpty(invalidPostcodes)) {
+            final List<String> postcodeCreated = adminService.addCourtPostcodes(slug, postcodes);
+            return created(URI.create(StringUtils.EMPTY)).body(postcodeCreated);
+        }
+        throw new InvalidPostcodeException(invalidPostcodes);
+    }
+
+    /**
+     * Deletes existing postcodes currently handled by the court for civil service type
+     * @param slug Court slug
+     * @param postcodes a list of postcodes to be deleted
+     * @return Nothing if successful, else return a list of invalid postcodes and a '400' response
+     */
+    @DeleteMapping(path = "/{slug}/postcodes")
+    @ApiOperation("Delete existing court postcodes")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successful", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 400, message = "Invalid postcodes", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role(FACT_SUPER_ADMIN)
+    public ResponseEntity deleteCourtPostcodes(@PathVariable String slug, @RequestBody List<String> postcodes) {
+        final List<String> invalidPostcodes = validationService.validatePostcodes(postcodes);
+        if (CollectionUtils.isEmpty(invalidPostcodes)) {
+            adminService.deleteCourtPostcodes(slug, postcodes);
+            return ok().build();
```

Comment:
Do we want to return alongside the request the list of resources that have been deleted? 

I think its best practices to return either the list of resources deleted, or the amount. Can't remember which! To google !

Reply:
Don't think we should do that for DELETE endpoint. For PUT and POST they are fine to return what have been added/created (although it is a long standing debatable issue). If you look that spring boot data JPA delete methods, they don't return any object after deletion. So to return something from the DELETE endpoint, we need to either return what has ben passed in, or get the current content from the database to be returned. This can be achieved by simply calling the GET endpoint.